extensions/MongoDbAtlas/MongoDbAtlas/MongoDbAtlasDatabaseHelper.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using MongoDB.Driver;
+
+namespace Microsoft.KernelMemory.MongoDbAtlas;
+
+/// <summary>
+/// Allow keeping a singleton for IMongoDatabase and MongoClient
+/// </summary>
+internal static class MongoDbAtlasDatabaseHelper
+{
+    private static MongoClient? s_client = null;
+
+    internal static IMongoDatabase GetDatabase(string connectionString, string databaseName)
```

Comment:
I changed the code, actually I've ported a code from other projects of mine, client is not static anymore but it is stored inside a static concurrent dictionary. This will ensure that we will create a single IMongoClient for each different connection string. At the same time we allow to use 2 different connections, one for storage and other for memory. 

Also used the same class from the SearchHelper class.